dbt/adapters/teradata/connections.py

```diff
@@ -288,6 +331,47 @@ def connect():
         try:
             connection.handle = teradatasql.connect(**kwargs)
             connection.state = 'open'
+
+            # checking if query_band connection parameter exists. If it does -> executing the set query_band sql
+            if credentials.query_band:
+                try:
+                    cursor = connection.handle.cursor()
+
+                    query_band_txt = credentials.query_band
+
+                    if query_band_txt[-1] != ';':
+                        query_band_txt += ';'
+
+                    """checking if appname= key exist in query_band and if dbt exist as value of 
+                    that key otherwise appending DBT in the end """
```

Comment:
The logic is duplicated. Can we move to single place like to a method and call in required places to avoid maintenance issues.

Reply:
Create a function apply_query_band() for reusability.
Thank you for the suggestion!